src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Windows.cs

```diff
@@ -412,5 +420,29 @@ internal ThreadPoolBoundHandle GetOrAllocateThreadPoolBoundHandleSlow()
             bool trySkipCompletionPortOnSuccess = !(CompletionPortHelper.PlatformHasUdpIssue && _protocolType == ProtocolType.Udp);
             return _handle.GetOrAllocateThreadPoolBoundHandle(trySkipCompletionPortOnSuccess);
         }
+
+        private static bool IsReuseUnicastPortSupported()
+        {
+            Interop.Winsock.EnsureInitialized();
+
+            try
+            {
+                IntPtr socket = Interop.Winsock.WSASocketW(AddressFamily.InterNetwork, SocketType.Stream, 0, IntPtr.Zero, 0, (int)Interop.Winsock.SocketConstructorFlags.WSA_FLAG_NO_HANDLE_INHERIT);
+                using var handle = new SafeSocketHandle(socket, ownsHandle: true);
+                if (handle.IsInvalid)
+                    return false;
+
+                int optionValue = 1;
+                SocketError error = Interop.Winsock.setsockopt(
+                    handle,
+                    SocketOptionLevel.Socket,
+                    SocketOptionName.ReuseUnicastPort,
+                    ref optionValue,
+                    sizeof(int));
+                return error == SocketError.Success;
+            }
+            catch { }
```

Comment:
I used `try-catch` block because `SocketProtocolSupportPal` does the same:

https://github.com/dotnet/runtime/blob/f3bad4f82787f3f2f0e949c750cd3c6f78705468/src/libraries/Common/src/System/Net/SocketProtocolSupportPal.Windows.cs#L18-L38

Don't know what can potentially throw here.

Reply:
For native calls like this, usually the only thing that can throw is ObjectDisposedException if the SafeHandle is disposed. Maybe if it's invalid too.

That should never be the case here, but there's no harm in the try/catch just in case.